source/Octopus.Tentacle/Kubernetes/IKubernetesPodLogService.cs

```diff
@@ -83,16 +87,28 @@ public KubernetesPodLogService(
                 var sinceTime = scriptPodSinceTimeStore.GetPodLogsSinceTime(scriptTicket);
                 try
                 {
-                    return await GetPodLogsWithSinceTime(sinceTime);
+                    try
+                    {
+                        return await GetPodLogsWithSinceTime(sinceTime);
+                    }
+                    catch (UnexpectedPodLogLineNumberException ex)
+                    {
+                        var message = $"Unexpected Pod log line numbers found with sinceTime='{sinceTime}', loading all logs";
+                        tentacleScriptLog.Verbose(message);
+                        Log.Warn(ex, message);
+
+                        //If we somehow come across weird/missing line numbers, try load the whole Pod logs to see if that helps
+                        return await GetPodLogsWithSinceTime(null);
+                    }
                 }
-                catch (UnexpectedPodLogLineNumberException ex)
+                catch (PodLogEncryptionKeyException ex)
                 {
-                    var message = $"Unexpected Pod log line numbers found with sinceTime='{sinceTime}', loading all logs";
+                    //if we can't read the pod log encryption key for a while
+                    var message = $"Failed to read pod log encryption key. No new pod logs will be read.";
```

Comment:
At a minimum, we still get whether the script exited successfully from K8s, so this is a reasonable trade-off to protect against transient decryption problems.